java/code/src/com/suse/manager/webui/websocket/Notification.java

```diff
@@ -59,23 +59,23 @@
      */
     @OnOpen
     public void onOpen(Session session, EndpointConfig config) {
-        User user = (User)session.getUserProperties().get("currentUser");
         if (session != null) {
-            if (user != null) {
-                LOG.debug(String.format("Hooked a new websocket session [id:%s]", session.getId()));
-                handshakeSession(user, session);
-
-                // update the notification counter to the unread messages
-                try {
-                    sendMessage(session, String.valueOf(UserNotificationFactory.unreadUserNotificationsSize(user)));
-                }
-                finally {
-                    HibernateFactory.closeSession();
-                }
-            }
-            else {
-                LOG.debug("no authenticated user.");
-            }
+            Optional.ofNullable(session.getUserProperties().get("webUserID"))
+                    .map(webUserID -> (Long) webUserID)
+                    .ifPresentOrElse(userId -> {
+                         LOG.debug(String.format("Hooked a new websocket session [id:%s]", session.getId()));
+                        handshakeSession(userId, session);
+
+                        // update the notification counter to the unread messages
+                        try {
+                            sendMessage(session,
+                                    String.valueOf(UserNotificationFactory.unreadUserNotificationsSize(userId)));
```

Comment:
Style nitpick: evaluate statically importing `valueOf` and `unreadUserNotificationsSize` to get this on one line :wink: 

I am maybe an addict to static imports of methods, and typically also add `ofNullable`, etc.

Do not feel obliged to change your patch in this sense - this is just to make you aware of this option and develop your own taste.

Reply:
I Like the simplicity of static imports, and it reduces the boilerplate code. But most of the times I don't like the readability of it. For example, in this case, if I'm statically importing the valueOf, it will be more difficult to know right away to which type in converting the value.
Where I really love the static import is on the automatic tests: asserts, mocks, etc, where the method names give at least one clue about what is happening :)
The `ofNullable` would also be a great candidate to static import since we can infer what is happening from the method name :)